shared/health/src/main/java/io/pravega/shared/health/HealthDaemon.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.health;
+
+
+/**
+ * The {@link HealthDaemon} is responsible for regularly updating the {@link Health} of the {@link HealthService}.
+ * This is useful in cases where health information is not regularly queried by some client. In the event of a crash or failure,
+ * (should we want to provide this information) it allows us to place an upper bound on how stale this {@link Health}
+ * information may be.
+ */
+public interface HealthDaemon {
```

Comment:
We don't use the name "Daemon" anywhere else. Mind if you rename this to `HealthServiceUpdater` or something similar to that, which clearly indicates what its role is?